src/iterators/iterator.md

```diff
@@ -2,49 +2,63 @@
 minutes: 5
 ---
 
-# `Iterator`
+# `Iterator` Trait
 
-The [`Iterator`][1] trait supports iterating over values in a collection. It
-requires a `next` method and provides lots of methods. Many standard library
-types implement `Iterator`, and you can implement it yourself, too:
+The [`Iterator`][1] trait defines how an object can be used to produce a
+sequence of values. For example, if we wanted to create an iterator that can
+produce the elements of a slice it might look something like this:
 
 ```rust,editable
-struct Fibonacci {
-    curr: u32,
-    next: u32,
+struct SliceIter<'a> {
+    slice: &'a [i32],
+    i: usize,
 }
 
-impl Iterator for Fibonacci {
-    type Item = u32;
+impl<'a> Iterator for SliceIter<'a> {
+    type Item = &'a i32;
```

Comment:
Minor detail, but I'd like to get away from the idea that all lifetimes are named `'a` -- maybe use `'sl` instead?

Reply:
How about `'s`?

Reply:
`'s` is good!